crates/top/re_sdk/src/lenses/ast.rs

```diff
@@ -846,4 +893,281 @@ mod test {
         assert_eq!(my_timeline.times_raw()[3], 130);
         assert_eq!(my_timeline.times_raw()[4], 140);
     }
+
+    // Helper function to create test data: list of structs with {timestamp: i64, value: String}
+    fn create_test_struct_list() -> ListArray {
+        use arrow::array::Int64Builder;
+
+        let mut struct_list_builder = ListBuilder::new(StructBuilder::new(
+            [
+                Arc::new(Field::new("timestamp", DataType::Int64, true)),
+                Arc::new(Field::new("value", DataType::Utf8, true)),
+            ],
+            vec![
+                Box::new(Int64Builder::new()),
+                Box::new(StringBuilder::new()),
+            ],
+        ));
+
+        let mut timestamp_counter = 1i64..;
+
+        // Row 0: [{1, "one"}, {2, "two"}, {3, "three"}]
+        struct_list_builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(timestamp_counter.next().unwrap());
+        struct_list_builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("one");
+        struct_list_builder.values().append(true);
+
+        struct_list_builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(timestamp_counter.next().unwrap());
+        struct_list_builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("two");
+        struct_list_builder.values().append(true);
+
+        struct_list_builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(timestamp_counter.next().unwrap());
+        struct_list_builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("three");
```

Comment:
Is there anything we can do to make this code more readable, e.g. by creating some helpers in `re_arrow_utils`? 🤔

Reply:
I'd love that, especially for those tests. I found this: https://docs.rs/arrow/latest/arrow/#serde-compatibility. Giving it a shot.